### DISCUSSION TITLE
Is it possible to disable environment variable tweaking at install time?

### DISCUSSION THREAD

**Discussion by pedoc** [ScoopInstaller/Scoop#5612] [Help and Questions] [answered]
Hi,everyone

When installing some software through scoop, they will modify the PATH environment variable, such as mpv-git
Now, I manually added the installation location of scoop in the environment variable, and don't want to modify it again when scoop is installed to avoid overwriting my settings
Is this possible? Thanks

As shown in the figure, there is a duplicate path here
![image](https://github.com/ScoopInstaller/Scoop/assets/6604230/a344b130-043f-4ce7-8c3d-aac595370cf5)

**Comment by HUMORCE** [comment]
Create your own bucket for maintain your preferences.

**Comment by pedoc** [comment]
> Create your own bucket for maintain your preferences.

Hi,
First,Thank you for your suggestion.

Then,My motivation for doing this is simple, because the ``PATH`` environment variable will quickly approach its maximum length as the amount of software installed gradually increases.
If we use some shorter environment variables for replacement, the length of the ``PATH`` variable can be effectively reduced, such as here
``%SCOOP_APPS%`` equal to ``%USERPROFILE%/scoop/apps``

So I'm expecting 2 options
1. Specify a parameter at installation time to prohibit adjustments to the ``PATH`` environment variable, because I have done this work manually
2. Support global configuration of scoop, define the root directory of scoop as an independent environment variable, and reference it in ``PATH``
For example
``scoop Root Directory=%SCOOP_ROOT%=%USERPROFILE%/scoop/``

Of course, I can maintain an independent bucket by myself, but this will obviously increase the workload. I need to copy the definition in the main bucket first, then modify the definition to remove the env_add_path setting, and finally manually modify the PATH variable, which is very troublesome

**Reply by rashil2000** [reply]
See https://github.com/ScoopInstaller/Scoop/issues/1813

**Comment by HUMORCE** [comment, marked as answer]
You always facing this problem, when a certain number of manifests are installed.

- Scoop already minimize PATH by shim as much as possible.
- Scoop suggests enabling **Long Paths** to solve this problem. `scoop checkup`

https://github.com/ScoopInstaller/Scoop/blob/f93028001fbe5c78cc41f59e3814d2ac8e595724/lib/diagnostic.ps1#L40-L55

See also: https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry
